### PR TITLE
new lambda function for creating a company

### LINKED ISSUE #4: save company details lambda

### PR DESCRIPTION
**Pull request by petarblazevski**
Closes #4 and #2

### CODE REVIEW

File: libs/response-lib.js [hunk 1]
```diff
@@ -6,6 +6,10 @@ export function failure(body) {
   return buildResponse(500, body);
 }
 
+export function validationError(body) {
+  return buildResponse(400, { errors: body });
```
Comment (IAfanasovMob) [inline libs/response-lib.js]: seems like [422](https://greenbytes.de/tech/webdav/rfc4918.html#rfc.section.11.2)

File: typings/company.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+/**
+ * @typedef {object} Address
+ * @property {string} city
+ * @property {string} houseNumber
+ * @property {string} street
+ * @property {string} zipCode
+ * @property {number} latitude
+ * @property {number} longitude
+ */
+
```
Comment (IAfanasovMob) [inline typings/company.js:23]: I like it! 👍

File: validation/createCompanySchema.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  companyLastName: Joi.string().required(),
+  iban: Joi.string().required(),
+  address: Joi.object({
+    city: Joi.string().required(),
+    houseNumber: Joi.string().required(),
+    street: Joi.string().required(),
+    zipCode: Joi.string().required(),
+    latitude: Joi.number().required(),
+    longitude: Joi.number().required()
+  })
```
Comment (IAfanasovMob) [inline validation/createCompanySchema.js:34]: I can not find a field for the KvK number. It is 8 digit legal code of company

Comment (petarblazevski) [reply validation/createCompanySchema.js:34]: I was looking at the .NET project, and it does not have this property as well. I will add it here

Comment (petarblazevski) [reply validation/createCompanySchema.js:34]: Funny enough, but Joi doesn't have a good solution for validating a number length of 8 :(

File: validation/createCompanySchema.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  iban: Joi.string().required(),
+  kvk: Joi.number()
+    .integer()
+    .min(10000000)
+    .max(99999999)
+    .required()
+    .messages({
+      'number.min': '"kvk" should be 8 digits long',
+      'number.max': '"kvk" should be 8 digits long'
+    }),
```
Comment (petarblazevski) [inline validation/createCompanySchema.js:26]: Not beautiful, but this is how Joi validates numbers. I just had to override the message,

**Review by IAfanasovMob — APPROVED** [review]
great wrok!

**Pull request merged**